### PR TITLE
sorter: add close check to AddEntry

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- Cancelling Unified Sorter's `Run` does not cancel `AddEntry`. This can cause unexpected channel congestion.

### What is changed and how it works?
- Added close detection in `AddEntry`.

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### Release note

- No release note

### CODE REVIEW

**Review by leoppro — APPROVED** [review]

File: cdc/puller/sorter/unified_sorter.go [hunk 1]
```diff
@@ -41,6 +41,8 @@ type UnifiedSorter struct {
 	dir         string
 	pool        *backEndPool
 	metricsInfo *metricsInfo
+
+	closeCh chan struct{}
```
Comment (amyangfei) [inline cdc/puller/sorter/unified_sorter.go:45]: Could you add some comments to describe the exit workflow of a unified sorter and why cancelling Unified Sorter's Run does not cancel AddEntry, such as the hierarchy of context used in `Run` and `AddEntry`

**Review by amyangfei — APPROVED** [review]

File: cdc/puller/sorter/unified_sorter.go (hunk 1 above)
Comment (liuzix) [reply cdc/puller/sorter/unified_sorter.go:45]: This is not a problem of the sorter itself, but of the caller(s). There is no guarantee that the contexts used to call `Run` and `AddEntry` are the same one.

Comment (amyangfei) [reply cdc/puller/sorter/unified_sorter.go:45]: Yes, what I mean is there is no limit of the context hierarchy here, please add comments to address it

**Pull request merged**